torch/csrc/jit/codegen/cuda/kernel_cache.cpp

```diff
@@ -266,61 +275,62 @@ FusionExecutorCache::FusionExecutorCache(std::unique_ptr<Fusion>&& fusion)
   FUSER_PERF_SCOPE("FusionExecutorCache::FusionExecutorCache");
   // avoid putting `has_reduction_` in the initializer list
   has_reduction_ = fusion_->hasReduction();
+
+  if (has_reduction_) {
+    FusionGuard fg(fusion_.get());
+
+    // Use dependency check to find the reduction tv as it returns used values
+    // instead of exprs.
+
+    // The call is relatively heavy weight, consider caching
+    auto used_vals = DependencyCheck::getAllValsBetween(
+        {fusion_->inputs().begin(), fusion_->inputs().end()},
+        fusion_->outputs());
+
+    // Find the reduction tensor view, make sure there's only one
+    for (auto val : used_vals) {
+      if (val->getValType().value() == ValType::TensorView) {
+        auto tv = val->as<TensorView>();
+        if (tv->hasReduction()) {
+          TORCH_INTERNAL_ASSERT(
+              reduction_tv_ == nullptr,
+              "Already found a reduction tensorview, cannot handle fusion of multiple reductions.");
+          reduction_tv_ = tv;
+        }
+      }
+    }
+
+    TORCH_INTERNAL_ASSERT(
+        reduction_tv_ != nullptr,
+        "Could not find the reduction tensor view in the fusion.");
+  }
 }
 
 std::vector<at::Tensor> FusionExecutorCache::runFusionWithInputs(
     const at::ArrayRef<IValue>& inputs) {
   FUSER_PERF_SCOPE("runFusionWithInputs");
 
+  LaunchParams launch_params;
+
   // get unique id `unique_id` for given input set `inputs`;
   auto id_lookup_ret = inputs_id_lookup_.lookupId(inputs);
   if (id_lookup_ret.eviction) {
     evictCache(id_lookup_ret.evict_id);
   }
 
-  const size_t unique_id = id_lookup_ret.id;
-  const int device_index = getCommonDeviceCUDA(inputs);
+  size_t unique_id = id_lookup_ret.id;
```

Comment:
what is the reason for removing const?

Reply:
Good catch! I'll revert it.

I was doing measurement in with our tracing earlier and updated the value in a separate block.